python/cuml/cuml/fil/fil.pyx

```diff
@@ -659,14 +650,14 @@ class ForestInference(Base, CMajorInputTagMixin):
 
         if self.treelite_model is not None:
             if isinstance(self.treelite_model, treelite.Model):
-                treelite_model_bytes = self.treelite_model.serialize_bytes()
+                tl_model = self.treelite_model
             elif isinstance(self.treelite_model, bytes):
-                treelite_model_bytes = self.treelite_model
+                tl_model = treelite.Model.deserialize_bytes(self.treelite_model)
             else:
                 raise ValueError("treelite_model should be either treelite.Model or bytes")
             impl = ForestInference_impl(
                 self.handle,
-                treelite_model_bytes,
+                tl_model,
```

Comment:
Users (and `RandomForest*`) can create a `ForestInference` model from either a `treelite.Model`, or the serialized bytes version.

Before, creating a `ForestInference` model from `treelite.Model` would result in:
- Serializing the treelite model to bytes (using the python api)
- Deserializing the treelite model from bytes (using the C api)
- Converting the treelite model to a FIL model
- Freeing the temporary treelite model

None of that roundtripping is necessary, and an undue cost. We now just use the original `treelite.Model` directly if possible, or load the bytes as a `treelite.Model` (using the python API) otherwise. This is simpler, and removes a memory management footgun where an error earlier could result in the temporary treelite model not being properly freed.

Reply:
Nevermind, this isn't possible today due how we manage `libtreelite` on wheels. On conda, both `treelite` and `cuml` share the same `libtreelite` binary, meaning sharing handles between the python libraries is fine. For wheels though `cuml` and `treelite` have separate versions of `libtreelite`, which can cause problems. I've reverted this change for now (though I'm not happy about it).